Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -1942,6 +1923,49 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncFill, (JSGlobalObject* globalObject, Call
     return JSValue::encode(thisObject);
 }
 
+JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncToReversed, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto thisValue = callFrame->thisValue().toThis(globalObject, ECMAMode::strict());
+    RETURN_IF_EXCEPTION(scope, { });
+    if (UNLIKELY(thisValue.isUndefinedOrNull()))
+        return throwVMTypeError(globalObject, scope, "Array.prototype.fill requires that |this| not be null or undefined"_s);
+    auto* thisObject = thisValue.toObject(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    uint64_t length = toLength(globalObject, thisObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    if (UNLIKELY(length > std::numeric_limits<uint32_t>::max())) {
+        throwRangeError(globalObject, scope, "Array length must be a positive integer of safe magnitude."_s);
```

Comment:
Not sure I understand why the type error message mentions the function name, but this does not. Do we have a consistent style? Is "safe magnitude" a good phrase? Is it the term used in the JavaScript specification?

Reply:
This is to make error messages prior to this change consistent. The screenshot is taken in the latest Safari( this is defined at https://github.com/WebKit/WebKit/blob/a10d2e295345449f0213e2141054377d23a6c41f/Source/JavaScriptCore/runtime/ArrayConstructor.cpp#L40 )

![Screenshot 2024-12-07 at 13 39 03](https://github.com/user-attachments/assets/7ed011a8-0afe-4f6f-b544-75bb0539a9b9)